src/routes/suggestions/index.svelte

```diff
@@ -0,0 +1,298 @@
+		count: _.getPath('1.doc_count')
+	});
+
+	const mapResponseToFieldCount = _.pipe([
+		_.getPath('aggregations.messages.buckets'),
+		_.pairs,
+		_.mapWith(extractCountInfo),
+		_.sortWith([_.getKey('count')]),
+		_.reverse
+	])
+
+	function computeSearchQuery (query) {
+		return {
+			query: {
+				term: {
+					[selectedFieldName]: query
+				}
+			}
+		};
+	}
+
+	function computeFieldForCountQuery (name, term) {
+		return {
+			term : {
+				[name] : term
+			}
+		};
+	}
+
+	function computeCountQuery (searchTerm) {
+		const schema = getSchema(dataset);
+		const keywordFields = getKeywordFields(schema);
+		const filters = keywordFields.map(i => [i, computeFieldForCountQuery(i, searchTerm)]);
+
+		const query = {
+			size: 0,
+			aggs : {
+				messages : {
+					filters : {
+						filters : _.fromPairs(filters)
+					}
+				}
+			}
+		};
+		return query;
+	}
+
+	function computeDetailsQuery (requestedFieldName, term) {
+		return {
+			size: 0,
+			query: {
+				term: {
+					[requestedFieldName]: term
+				}
+			},
+			aggs: {
+				[requestedFieldName]: {
+					significant_text: {
+						field: requestedFieldName,
+						include: stopWords,
+						min_doc_count: 1
+					}
+				}
+			}
+		};
+	}
+
+	async function sendRequest (data) {
+		try {
+			return await request('POST', url, {data});
+		}
+		catch (error) {
+			return error.jsonMessage;
+		}
+	}
+
+	let userSelection;
+	function computeSelection (counts) {
+		const exists = counts.some(item => item.name === userSelection);
+		if (!userSelection || !exists) {
+			if (fieldCounts.length > 0) {
+				return fieldCounts[0].name;
+			}
+			return null;
+		}
+		return userSelection;
+	}
+
+	async function sendSearchRequest (event) {
+		response = null;
+		awaitingResponse = true;
+		const data = computeSearchQuery(event.detail);
+		response = await sendRequest(data);
+		awaitingResponse = false;
+	}
+
+	async function sendCountRequest () {
+		const data = computeCountQuery(searchValue);
+		const countResponse = await sendRequest(data);
+		// TODO check for error messages
+		const fullFieldCounts = mapResponseToFieldCount(countResponse);
+		fieldCounts = fullFieldCounts.filter(field =>
+			field.count > 0 || field.name === userSelection
+		);
+		selectedFieldName = computeSelection(fieldCounts);
+		console.log("fieldCounts", fieldCounts);
+	}
+
+	function sendIfTimeElapsed () {
+		const now = Date.now();
+		if (now - lastScheduling > SEND_DELAY) {
+			sendCountRequest();
+		}
+	}
+
+	function scheduleCountRequest (event) {
+		lastScheduling = Date.now();
+		searchValue = event.detail.trim().toLowerCase();
+		setTimeout(sendIfTimeElapsed, SEND_DELAY + 10);
+	}
+
+	async function sendSuggestionsRequest (event) {
+		const fieldName = event.detail;
+		const info = fieldCounts.find(fieldInfo => fieldInfo.name === fieldName)
+		if (info.suggestions) {
+			return;
+		}
+		info.suggestions = ['waiting...'];
+		fieldCounts = fieldCounts;
+		const data = computeDetailsQuery(fieldName, searchValue);
```

Comment:
I'd use mose specific names for both `data` and `computeDetailsQuery` (we shouldn't have to read a function's code to know what it does)